### PR TITLE
Update xcodebuild -showBuildSettings step to derive the build directory using xctool-args

### PR DESCRIPTION
**Pull request by paulstringer**
Under certain xctool-args constructs it was possible that when using showBuildSettings a different build directory would be returned to the one where the results of xctool test run were. E.g. using -destination in xctool-args would result in a different DerivedData build folder. This would mean when running 'arc unit' this would manifest itself as the error "Directory does not exist".

The fix here is to parse the args from .xctool-args and supply these to xcodebuild showBuildSettings to resolve the differences in DerivedData build folders. To make a non breaking change but to support xctool-arg configurations that include -destination or -arch (e.g. for selecting and iPad for Unit Tests), it conditionally only adds the -arch i386 if both these two flags are missing from xctool-args.

### CODE REVIEW

**Comment by mbishopim3** [comment]
Can you run `arc lint` on this? There's some PHP style items the `ComprehensiveLintEngine` enforces that would affect this code.

Otherwise we're trying this locally.

**Comment by paulstringer** [comment]
I'll check that out, ironically I didn't lint the actual new code :/ so I'll do that.

File: libcassowary/src/unit/MobileUnitTestEngine.php [hunk 1]
```diff
@@ -399,6 +402,23 @@ private function parseiOSOutput($test_results) {
         return $result_array;
     }
 
+	// Retrieve Args from the xctool-args file
+	private function xcodebuild_args() {
+		$xctoolargs_path = $this->projectRoot . "/.xctool-args";
```
Comment (mbishopim3) [inline libcassowary/src/unit/MobileUnitTestEngine.php]: This appears to be a required file which isn't always true either. Needs to have some fallback support.

**Comment by paulstringer** [comment]
Added a fallback for if xctool-args file is missing which will then add the minimum -sdh iphonesimulator and -arch i386 args. 

This also includes some auto fixes applied when running arc lint which has fixed some previous lint issues specifically around style of quotation. The new code all lints successfully.

**Comment by mbishopim3** [comment]
Works for us and looks good, thanks.

**Pull request merged**